ansible_collections/ds/ansible_ds/plugins/module_utils/dsentities.py

```diff
@@ -0,0 +1,904 @@
+
+    def addModifier(self, dn, type, attr, val):
+        dict = self._cfgMods
+        DiffResult.addModifier(dict, dn, type, attr, val)
+
+
+class YAMLHost(MyYAMLObject):
+    yaml_tag = u'!ds389Host'
+
+    OPTIONS = (
+        SpecialOption('prefix', 1, "389 Directory Service non standard installation path" ),
+    )
+    CHILDREN = ( 'instances', )
+
+    def __init__(self):
+        super().__init__(socket.gethostname())
+        self.prefix = self.getPath('{prefix}')
+
+    def MyPathNames(self):
+        return { 'hostname' : self.name, 'prefix' : os.environ.get('PREFIX', "") }
+
+    def getFacts(self):
+        ### Lookup for all dse.ldif to list the instances
+        for f in glob.glob(f'{self.prefix}/etc/dirsrv/slapd-*/dse.ldif'):
+            ### Extract the instance name from dse.ldif path
+            m = re.match(f'.*/slapd-([^/]*)/dse.ldif$', f)
+            ### Then creates the Instance Objects
+            instance = YAMLInstance(m.group(1), parent=self)
+            self.instances.append(instance)
+            ### And populate them
+            instance.getFacts()
+
+    def _prefixAction(self=None, action=None, action2perform=None):
+        option = action.option
+        if action2perform == OptionAction.DESC:
+            return f"Set PREFIX environment variable to {action.vto}"
+        elif action2perform == OptionAction.DEFAULT:
+            return os.environ.get('PREFIX', "")
+        elif action2perform == OptionAction.FACT:
+            return os.environ.get('PREFIX', "")
+        elif action2perform == OptionAction.CONFIG:
+            val = action.getValue()
+            log.debug(f"Instance: {action.target.name} config['slapd'][{option.name}] = {val} target={action.target}")
+            action.target._infConfig['slapd'][option.name] = val
+        elif action2perform == OptionAction.UPDATE:
+            setattr(action.facts, option.name, action.vto)
+            os.environ.set('PREFIX', action.vto)
+
+
+class YAMLInstance(MyYAMLObject):
+    yaml_tag = u'!ds389Instance'
+    LDBM_CONFIG_DB = 'cn=config,cn=ldbm database,cn=plugins,cn=config'
+    PARAMS = {
+        ** MyYAMLObject.PARAMS,
+        'started' : "Boolean to tell whether the server should be started or stopped.",
+        'dseMods' : "List of change that needs to be applied on dse.ldif after the instance creation",
+    }
+    CHILDREN = ( 'backends', )
+    OPTIONS = (
+        ConfigOption('backup_dir', 'nsslapd-bakdir', 'cn=config', None, "Desc" ),
+        ConfigOption('bin_dir', 'nsslapd-bin_dir', 'cn=config', None, "Desc" ),
+        ConfigOption('cert_dir', 'nsslapd-certdir', 'cn=config', None, "Desc" ),
+        ConfigOption('config_dir', 'nsslapd-config_dir', 'cn=config', None, "Desc" ),
+        ConfigOption('data_dir', 'nsslapd-data_dir', 'cn=config', None, "Desc" ),
+        ConfigOption('db_dir', 'nsslapd-db_dir', 'cn=config', None, "Desc" ),
+        ConfigOption('db_home_dir', 'nsslapd-db_home_dir', 'cn=config', None, "Desc" ),
```

Comment:
I think it's a work in progress here and you know about that but I'll mention it just in case. :)

These variables are not aligned with what we have in cn=config (most of the OPTIONS list).

For example, here, it should be `nsslapd-db-home-directory` and it should belong to `cn=bdb,cn=config,cn=ldbm database,cn=plugins,cn=config`

Reply:
Good catch ! I just made another pass on the options to clean up thing